### ISSUE TITLE
Intercom integration is failing in some cases

### ISSUE THREAD

**Issue by synaptiko** [segmentio/analytics-next#411]
We are using Segment together with Intercom integration but sometimes (specifically on our login page) the initialization is failing. We weren't able to find a fix as we really don't understand why it's happening in the first place. On all other pages it works fine.

Can you please help us to either debug the problem or provide some instructions how to fix it? Thank you.

Here's the error we see, I can't see anything related to our code, that's why it's hard to debug:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'hideDefaultLauncher')
    at y (index.js:39:3)
    at e.OJRb.l.bootOrUpdate (index.js:220:17)
    at e.OJRb.l.page (index.js:82:33)
    at e.page (protos.js:343:17)
    at protos.js:204:23
    at c (index.js:64:9)
    at t.exports (index.js:122:59)
    at e.flush (protos.js:203:3)
    at e.<anonymous> (index.js:21:15)
    at e.r (index.js:65:8)
    at e.n.emit (index.js:145:20)
    at e.ready (protos.js:307:8)
    at index.js:21:15
    at protos.js:262:24
    at HTMLScriptElement.<anonymous> (index.js:27:47)
```

**Comment by synaptiko** [comment]
Update: it looks like we are calling `trackPage` somehow too early. But even if we delay it, it's still happening sometimes and sometimes not. But we also call `identify` just before it and that one doesn't seem to be failing.

**Comment by synaptiko** [comment]
Here's a workaround I've implemented for now:
```ts
  if (state.enabled) {
    (async function() {
      await new Promise((resolve) => {
        const intervalId = setInterval(() => {
          if (window.Intercom?.booted) {
            clearInterval(intervalId);
            resolve();
          }
        }, 250);
      });

      window.analytics.page(pathname, context, adapterOptions);
    })();
  }
```

It's not the nicest thing but it seems to work.

**Comment by chrisradek** [comment]
@synaptiko 
Is the page where this is happening public by chance? I'd be happy to take a look if you can point me to a page where this is happening.

In the meantime we can check if there's a more reliable way to determine if Intercom is loaded versus checking that `window.Intercome` is a function.

**Comment by synaptiko** [comment]
@chrisradek Yes, you can see the error here: https://www.bayzat.com/auth/login

Thank you for looking into it.

**Comment by chrisradek** [comment]
@synaptiko Thanks for the link, I was able to narrow down what's triggering the error.

It looks like you have an `analytics.page` call where you pass in `options` that include the integrations to enable/disable. Something like this:

```js
function trackPage(integrations) {
  const options = integrations || { integrations: { Intercom: true } };
  analytics.page(`pathname`, { userId }, options);
}
```

When I debug this function, I see that `integrations` being passed into `trackPage` is already defined, and looks like this:
```js
{ All: false, Intercom: true }
```
The issue is that this is being passed directly as options to the `page` call, when it should be set as the `integrations` value in the options.

Basically, the error is happening when `page` is called like
```js
analytics.page('pathname', { userId }, { All: false, Intercom: true })
```
Changing it to
```js
analytics.page('pathname', { userId }, { integrations: { All: false, Intercom: true } })
```
fixes it for me locally.

I'm looking at the minified code so can't provide too much more context than that, but I think you just need to make sure that the `options` that get assigned in my `trackPage` example is what you expect.

**Comment by synaptiko** [comment]
@chrisradek You are right, we really had such an error in our code :man_facepalming: We didn't realize that because the error in the console is a bit cryptic :-) But I fixed that and tested, it seems to work now. Thank you very much. I think this issue can be closed now.

Note: I've introduced more strict type for `adapterOptions` on our side to prevent this to happen again, our previous type was too relaxed and hence we missed this.
